packages/react-native/src/Authenticator/hooks/useFieldValues/utils.ts

```diff
@@ -2,9 +2,18 @@ import { Logger } from 'aws-amplify';
 import { AuthenticatorRouteComponentName } from '@aws-amplify/ui-react-core';
 
 import { RadioFieldOptions, TypedField } from '../types';
+import { ContactMethod, UnverifiedContactMethods } from '@aws-amplify/ui';
 
 const logger = new Logger('Authenticator');
 
+export const supportedContactMethods: Record<
```

Comment:
Can this just be an array?

Reply:
The reason why I went with `Record<keyof UnverifiedContactMethods,ContactMethod>` is to ensure that if in the future another unverified contact method is added, or an existing one is changed we'll get some safety here.

Reply:
Totally align on the intention, but have a few callouts:
- creating an object with values that themselves do not correlate to any logic has the potential to confuse future maintainers since the existence of the values implies they have a use case, as objects are meant to store values for usage elsewhere, not type checking. Additionally, `ContactMethod` does not correlate to `UnverifiedContactMethod`, its use case is to denote the allowed types of sign up confirmation code destination
- usage of a `string` enum and utility would much better suit this use case, provide more utility in the codebase, and could be colocated with the `UnverifiedContactMethods` type (which we maintain as well)